### PR TITLE
Ngrok Tunnelling and NationBuilder Webhook Part 1

### PR DESCRIPTION
**Pull request by dkoodev**
## Ticket Information

**[Ngrok Tunnelling and NationBuilder Webhook Part 1](https://www.notion.so/ctc/Ngrok-Tunnelling-and-NationBuilder-Webhook-Part-1-779e196ee8b640d2b5f7116fd2686253)**

Added ngrok, created new post route for webhook, tested nationbuilder hitting webhook, and created the interface for a NationBuilderPerson.

## PR Checklist

- [ ] Wrote ava tests
- [ ] Reviewed by **all** @ctcusc/RiverLA team members

## Helpful Resources
- Be sure to start off by reading our Notion ticket. Quinn's a legend and writes super detailed tickets with lots of helpful links, so check that out.
- What is `ngrok`? A brief description can be found [here](https://ngrok.com/product "https://ngrok.com/product") and the NPM information can be found [here](https://www.npmjs.com/package/ngrok "https://www.npmjs.com/package/ngrok").
- Note that there's a part 2 to this ticket. If you want to read more about what's coming, you can find it [here](https://www.notion.so/ctc/Webhook-Part-2-Request-filtering-and-calling-API-clients-afe9cd5d65e84db99b34e861f94c48a9 "Webhook Part 2: Request filtering and calling API clients").
- If you need any more information, just message me on Slack! Thanks :)

### CODE REVIEW

**Comment by dkoodev** [comment]
@QuinnNgo This endpoint worked when using the test functionality in the webhook settings page, but when filling out the form, it doesn't seem to hit the webhook. Not sure why it won't work in the actual page , but we should look into it.

File: src/index.ts [hunk 1]
```diff
@@ -7,3 +10,12 @@ app.listen(env.server.port, () => {
...
 });
+
+(async function() {
+  const ngrokUrl = await ngrok.connect();
+  console.log(
+    `[Instructions to test webhook]
+    Go to https://larivercorp.nationbuilder.com/admin/webhooks/new and
+    add a new webhook with Endpoint URL: ${ngrokUrl}/webhooks/nationbuilder/personCreated.`,
+  );
+})();
```
Comment (QuinnNgo) [inline src/index.ts]: This should only happen when `process.env.NODE_ENV === 'development'`. Would you mind adding a `nodeEnv` to `/src/env.ts` and adding it here?

File: src/routers/webhooks.ts [hunk 2]
```diff
@@ -1,8 +1,24 @@
...
-  res.send('webhook route');
+router.use('/', (_, res, next) => {
+  next();
+});
+
+router.post('/nationbuilder/personCreated', function(req, res) {
+  const nationBuilderPerson: NationBuilderPerson = {
+    email: req.body.payload.person.email,
+    firstName: req.body.payload.person.first_name,
+    phone: req.body.payload.person.phone,
```
Comment (QuinnNgo) [inline src/routers/webhooks.ts]: nit: destructure this above:

```
const { email, first_name, phone } = req.body.payload.person;
```

File: src/routers/webhooks.ts [hunk 3]
```diff
@@ -1,8 +1,24 @@
...
+  };
+  console.log('NationBuilderPerson', nationBuilderPerson, 'Created');
```
Comment (QuinnNgo) [inline src/routers/webhooks.ts]: Remove log 👀

File: src/index.ts [hunk 4]
```diff
@@ -1,3 +1,6 @@
+// Import modules
+import ngrok = require('ngrok');
```
Comment (olivia-hong) [inline src/index.ts]: Change to `import ngrok from 'ngrok';`

Comment (yash-chandak) [reply src/index.ts]: Got it! Linking the StackOverflow post you sent me to keep up documentation

https://stackoverflow.com/questions/37651495/difference-between-import-from-and-import-require-in-typescript

File: src/routers/webhooks.ts (hunk 2 above)
Comment (yash-chandak) [reply src/routers/webhooks.ts]: Cool! Attaching the link you sent me for documentation:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

**Comment by QuinnNgo** [comment]
> @QuinnNgo This endpoint worked when using the test functionality in the webhook settings page, but when filling out the form, it doesn't seem to hit the webhook. Not sure why it won't work in the actual page , but we should look into it.

@dkoodev if you're repeatedly using the same email for testing, you may want to test using the `Person Changed` event instead? We'll have to do some testing to see how well this works.

**Pull request merged**